src/globus_sdk/services/auth/client_types/base.py

```diff
@@ -395,10 +417,13 @@ def oauth2_token(
         log.info("Fetching new token from Globus Auth")
         # use the fact that requests implicitly encodes the `data` parameter as
         # a form POST
+        data = dict(form_data)
+        if body_params:
+            data.update(body_params)
```

Comment:
FWIW, `utils.PayloadWrapper` supports `update()`, so I think you could skip the conversion. But also, this is not at all important. 🤷

Reply:
Actually the thought was to copy to not side-effect update `form_data`. Probably doesn't matter; just a defensive habit.

Reply:
Ah! A good habit.

When I do this (and I don't always, which is maybe a _bad_ habit), I comment on it even if it's just `# copy`. As evidenced here, I easily find it confusing otherwise. 😅